eng/Directory.Build.Data.targets

```diff
@@ -40,6 +40,14 @@
     </PackageReference>
   </ItemGroup>
 
+  <!-- Enable StyleCop  -->
+  <ItemGroup>
```

Comment:
I thought you planned to add a property that folks could set to condition out this package reference as a way to opt-out. Is there a reason that using the disabled ruleset file is better? It seems like a simple bool property would be simplier and easier.

Reply:
I tried <RunCodeAnalysis>false</RunCodeAnalysis> as well as <PackageReference> Exclude. Both of them did not do the Job. Also projects like Batch already have a rulesetfile for FxCop. So the only way I could combine them was using the other rulesetfile so I can reference one from the other.

Reply:
I would expect you to have a condition on the ProjectReference directly or the ItemGroup adding it similar to https://github.com/Azure/azure-sdk-for-net/blob/master/eng/Directory.Build.Data.targets#L29. 

If projects already have a ruleset file they will have to be handled specially no matter what we do. I suspect we should be able to merge multiple ruleset files, but we would need to figure out how to do that.

Reply:
Actually we already condition the other analyzer packages https://github.com/Azure/azure-sdk-for-net/blob/master/eng/Directory.Build.Data.targets#L35

Reply:
Longer term Batch is open to aligning on a single ruleset file - the ruleset that I have set up right now only enforces `ConfigureAwait(false)` must be used on all `await`'s because we've had numerous bugs in the past (years ago) where we had missed one and it caused issues for people in UI contexts.

I actually tried applying a more aggressive ruleset but there were a lot of issues I needed to fix so I delayed it. But we can have an item to align more closely.